### PR TITLE
[F11-3] Gamified Task Manager

### PR DESCRIPTION
**Pull request by jeffkwoh**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -1,961 +1,483 @@
...
+
+
+## Non-functional requirements
+- There should be no noticable lag when typing on the command line.
+- Should work on any mainstream OS as long as it has Java 9 or higher installed.
+- Storage of application data will be in a text file.
+- Should be able to hold up to 1000 tasks without a noticable decline in performance.
+- Should not take more than 3 seconds to complete a command.
+- System should be maintainable with proper documentation.
+- Proper feedback to user when error occurs. // @jelena: Usability
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: Remember to remove my name :)

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -1,961 +1,483 @@
...
+* 2. Application fetches data and displays statistics.
+#### Extensions
+* 2a. There are no statistics.
+    * 2a1. Application displays a flavour text explaining that the user needs to complete a task first to get statistics.
+
+### Use case: Failing to complete task
+#### MSS
+* 1. User fails to complete tasks on time
+* 2. Application displays points lost and current current points to user.
+### Extensions
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: I think there should be another `#` here

**Review by jelneo — COMMENTED** [review]
Good effort, continue working towards v1.1.

**Comment by jelneo** [comment]
Good effort in achieving v1.1 and please resolve the merge conflict.

Some general feedback:
- Travis build status badge should point to the build status of your own repo, not se-edu/AB-4. Other badges should be updated to match your project or removed
- Update this:
image
-  I think there is a problem with rendering the user guide and developer guide on netlify, please check it out

Feedback on About Us page:
- Please upload a higher quality image for Jeffrey
- Should assign the different components in the app to members

Feedback on User Guide:
- Include all the sections that the AB4 User Guide has. Refer to the AB4 User guide as an example. Remember to update the relevant portions in the guide to fit the application that you are developing
- You can include all the features/commands that you have brainstormed and then add ‘coming in v2.0’ for features that are not implemented yet.

Feedback on Developer Guide:
- Appendices should be at the bottom of the dev guide
- Spelling error:
image

**Comment by pengnam** [comment]
@unsatisfiedpopcorn @jiaxiu11 @remo5000 @chikchengyao 
I hope this tags us all.

**Comment by jelneo** [comment]
Good job on meeting milestone v1.2 @unsatisfiedpopcorn @pengnam @jiaxiu11 @remo5000 @chikchengyao 

Some general feedback:
- Please upload a better quality photo of Jeffrey in the About Us page. He looks really blurry here:
image
- User and dev guide not rendered properly on netlify

Some feedback on developer guide:
- developer guide is still not rendered properly on netlify
- the order of contents should follow AB-4 Dev guide. E.g. 
image with appendices at the bottom
- 3.1 Achievement mechanism
image
   1) Double-check if the circled relationships (AchievementRecord and ModelManager) are supposed to be an aggregation (I'm not saying they are wrong. It depends on how you design and implement it). It is mentioned that `AchievementRecord` 

>  stores the achievement information of the user internally with an Xp field and a Level field. When the task manager application runs for the first time, it is initialized at 0 xp and level 1, xp and level information in `AchievementRecord` are updated subsequently when tasks are completed.

Hence, I understood it as each `AchievementRecord` will always have `Xp` and `Level`.
    2) Rearrange and enlarge the `Level` box so that 'enumeration' does not spread across two lines
   
-------------------------------------------------------------------------------------------------------------
image


in red: arrange the elements in the diagram such that the words don't overlap and the text don't spill over to the next line
in blue: move the lifeline to the center of the box
optional box: the bars should be in front of the box and the 'opt' and condition shouldnot overlap with the activation bars
 
  -------------------------------------------------------------------------------------------------------------
image


in red: same comments as above about text formatting
The Ui and EventsCenter can switch positions so that it will be clearer that some external class is calling a method in EventsCenter

   -------------------------------------------------------------------------------------------------------------

image


in red: same comments as above about text formatting
It would be better if the flow of the process is from top to bottom instead of left to right.
   -------------------------------------------------------------------------------------------------------------
- 3.2 Batch Completion via Label feature
    1) Incorrect usage of `::`: `Label::many` . see this link: https://www.codementor.io/eh3rrera/using-java-8-method-reference-du10866vx
    2) It would be good to label your diagrams and then refer to it in the steps in the usage scenario
   -------------------------------------------------------------------------------------------------------------
image


correct grammar in picture above

   -------------------------------------------------------------------------------------------------------------
image
It would be good to rephrase the part pointed to by the arrow to make the explanation clearer.
Also, include diagrams that are taught in the module such as sequence diagrams
- 3.3 Model Rollback feature
Include diagrams that are taught in the module such as sequence diagrams
- 3.4 Dependency Feature Modelling
Possible to elaborate more using the sequence diagram shift it into the 'Current implementation' section.
Sequence diagram can be rearranged and spaced out such that text and other image elements don't overlap. Please save the diagram as an image before using it to remove the squiggly red lines under some words.
Align the arrows properly such that they touch and overshoot the activation bars.

**Comment by jeffkwoh** [comment]
@jelneo I tried not to have a high-resolution image of myself as much as I can. Apologies, I should have clarified this earlier, I pixelated the image according to the guidelines found in the cs2103 website:
> If you are uncomfortable posting your photo due to security reasons, you can post a lower resolution image...

I can replace with a more acceptable pixelated image if possible. I hope this is okay

**Comment by jelneo** [comment]
Some things to note:
- Make sure you have contextualize the dev guide according to your product.
- for the image below, is the underlined portion a typo?
image
- to be safe, add a return arrow for self-invocation:
image
- is there a repeated word here?
image
- for the image below:
1. make it into one line
2. & 5. the arrows overshoot the activation bars
3. missing arrowhead
4. create an outline for the stacked activation box. include a return arrow for it too.
image
- for the image below, ensure that the two ends of the arrow touches the activation bars (don't leave nay gaps). 
1. expand the box and shift model to the right so that the diagram does not look squished.
2. the arrowhead should be right at the start of the activation bar.
image
- create an outline for the stacked activation box and spread them out a bit vertically so it doesn't look cluttered:
image
- Lastly, take a look at my previous comments regarding your dev guide if you have not done so.

**Pull request closed without merging**